### DISCUSSION TITLE
Why are projects restored before getting built in the sample Dockerfiles?

### DISCUSSION THREAD

**Discussion by georgiosd** [dotnet/dotnet-docker#4187] [Q&A]
This seems to be assumed:

```
# after copying *.csproj
dotnet restore
# after copying source
dotnet build -c release --no-restore
```

Why do that, instead of directly building?

Is it purely to be able to cache the restore step?

**Comment by mthalman** [comment]
> Is it purely to be able to cache the restore step?

Yes, exactly. Consider the scenario of an iterative development process on your local machine where you build the Dockerfile. Once the restore has been executed the first time and you've not changed the content of the csproj since, you don't need to bother executing the restore again when you're building your code changes. It will be cached. This is nice because restore can take a fair amount of time depending on how many NuGet packages are involved.

**Reply by georgiosd** [reply]
Thank you, that makes sense. May be useful to add a note for that in the samples, for the less experienced folk like me :)
